server/routes/appointments/search/handlers/select-date.test.ts

```diff
@@ -0,0 +1,75 @@
+import { Request, Response } from 'express'
+import { plainToInstance } from 'class-transformer'
+import { validate } from 'class-validator'
+import SelectDateRoutes, { SelectDate } from './select-date'
+import { associateErrorsWithProperty } from '../../../../utils/utils'
+import { ServiceUser } from '../../../../@types/express'
+
+jest.mock('../../../../services/activitiesService')
+
+describe('Route Handlers - Appointments Management - Search Results', () => {
+  const handler = new SelectDateRoutes()
+  const user = { activeCaseLoadId: 'MDI', username: 'USER1', firstName: 'John', lastName: 'Smith' } as ServiceUser
+  let req: Request
+  let res: Response
+
+  beforeEach(() => {
+    req = {} as unknown as Request
+
+    res = {
+      locals: {
+        user,
+      },
+      render: jest.fn(),
+      redirect: jest.fn(),
+    } as unknown as Response
+
+    jest.resetAllMocks()
+  })
+
+  describe('GET', () => {
+    it('should render select date page', async () => {
+      await handler.GET(req, res)
+      expect(res.render).toHaveBeenCalledWith('pages/appointments/search/select-date')
+    })
+  })
+
+  describe('POST', () => {
+    it('should redirect with correct date', async () => {
+      req.body = {
+        startDate: '05-10-2021',
```

Comment:
Other usages of the SimpleDate component treat the req body parameter as an instance of SimpleDate then format the date using yyyy-MM-dd. I think your code is relying on the toString method on SimpleDate. I don't think that's wrong however it does mean that this test isn't exactly as the runtime will be. Maybe consider changing it to similar to the 'should save start date, start time and end time in session and redirect to repeat page' test in dateAndTime.test.ts?

The expect here should then be /appointments/search?startDate=10-5-2021 I think